samples/Ralltiir/Ralltiir.Server/Controllers/AccountController.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Ralltiir.Server.Models;
+using Ralltiir.Server.ViewModels;
+
+namespace Ralltiir.Server.Controllers
+{
+    [Authorize]
+    [ApiController]
+    [Route("[controller]")]
+    public class AccountController : Controller
+    {
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly SignInManager<ApplicationUser> _signInManager;
+
+        public AccountController(
+            UserManager<ApplicationUser> userManager,
+            SignInManager<ApplicationUser> signInManager)
+        {
+            _userManager = userManager;
+            _signInManager = signInManager;
+        }
+
+        [AllowAnonymous]
+        [HttpPost("register")]
+        public async Task<IActionResult> Register(RegisterUserRequest request)
+        {
+            var user = new ApplicationUser {UserName = request.Email, Email = request.Email};
+            
+            var result = await _userManager.CreateAsync(user, request.Password);
+            
+            if (result.Succeeded)
+            {
+                return Ok();
+            }
+
+            foreach (var error in result.Errors)
+            {
+                ModelState.AddModelError(string.Empty, error.Description);
+            }
+
+            return BadRequest(ModelState);
+        }
+
+        [AllowAnonymous]
+        [IgnoreAntiforgeryToken]
```

Comment:
Hum, we'll need to remove that and enforce antiforgery for this endpoint using `ValidateAntiForgeryToken`, as `PasswordSignInAsync` returns an authentication cookie. Not doing so would expose older browsers that don't support same-site to session fixation attacks.

Please feel free to add a comment to make sure people are aware of that.

Reply:
I finally figured out how to fix the antiforgery issue. 🎉